src/inmanta/resources.py

```diff
@@ -344,18 +333,26 @@ def validate(cls) -> None:
                     "Resource %s is a reserved keyword and not a valid field name, reported in %s" % (field, cls.__name__)
                 )
 
-    def __init__(self, _id: "Id") -> None:
+    def __init__(self, _id: "Id", fields: JsonType = None) -> None:
```

Comment:
I wonder whether the loading should move to a helper method instead of in the constructor? That would make overriding it less tricky. However then we need to find out why we possibly need to set all the fields to "none". Not if this is still necessary.

Reply:
good suggestion! thx